docs/reference/search/retriever.asciidoc

```diff
@@ -371,6 +374,118 @@ GET movies/_search
 ----
 // TEST[skip:uses ELSER]
 
+[[rescorer-retriever]]
+==== Rescorer Retriever
+
+The `rescorer` retriever re-scores only the results produced by its child retriever.
+For the `standard` and `knn` retrievers, the `window_size` parameter specifies the number of documents examined per shard.
+
+For compound retrievers like `rrf`, the `window_size` parameter defines the total number of documents examined globally.
```

Comment:
This distinction could be a bit technical/complex to understand (as the compound's retriever rewrite is transparent to the user) but I can see how we want it to be in line with what we already.

Reply:
Yep, I didn't want to add another round just to make it consistent with the compound execution.